src/main/java/eu/bbmri_eric/negotiator/database/model/Negotiation.java

```diff
@@ -140,15 +127,39 @@ public void setCurrentState(NegotiationState negotiationState) {
   }
 
   public void setStateForResource(String resourceId, NegotiationResourceState state) {
-    currentStatePerResource.put(resourceId, state);
-    if (!state.equals(NegotiationResourceState.SUBMITTED)) {
-      NegotiationResourceLifecycleRecord record =
-          NegotiationResourceLifecycleRecord.builder()
-              .changedTo(state)
-              .resource(lookupResource(getResources(), resourceId))
-              .build();
-      this.negotiationResourceLifecycleRecords.add(record);
+    NegotiationResourceLifecycleRecord record =
+        NegotiationResourceLifecycleRecord.builder()
+            .changedTo(state)
+            .resource(lookupResource(getResources(), resourceId))
+            .build();
+    this.negotiationResourceLifecycleRecords.add(record);
+  }
+
+  public Map<String, NegotiationResourceState> getCurrentStatePerResource() {
```

Comment:
please refactor this method as it is quite long. Divide it into multiple private methods

Reply:
Fixed in fae91486edd5eb4136965a0060278f579255ae20

Reply:
Migration script added in 97adafaf3cc969b82f50fa25f60fdbc2431e8e0